pkg/kubelet/winstats/perfcounters.go

```diff
@@ -0,0 +1,119 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package winstats
+
+import (
+	"errors"
+	"fmt"
+	"github.com/lxn/win"
+	"time"
+	"unsafe"
+)
+
+const (
+	CPUQuery                  = "\\Processor(_Total)\\% Processor Time"
+	MemoryPrivWorkingSetQuery = "\\Process(_Total)\\Working Set - Private"
+	MemoryCommittedBytesQuery = "\\Memory\\Committed Bytes"
+)
+
+// Metric collected from a counter
+type Metric struct {
+	Name  string
+	Value float64
+}
+
+// NewMetric construct a Metric struct
+func NewMetric(name string, value float64) Metric {
+	return Metric{
+		Name:  name,
+		Value: value,
+	}
+}
+
+func (metric Metric) String() string {
+	return fmt.Sprintf(
+		"Name: %s | Value: %s",
+		metric.Name,
+		metric.Value,
+	)
+}
+
+func readPerformanceCounter(counter string, sleepInterval int) (chan Metric, error) {
+
+	var queryHandle win.PDH_HQUERY
+	var counterHandle win.PDH_HCOUNTER
+
+	ret := win.PdhOpenQuery(0, 0, &queryHandle)
+	if ret != win.ERROR_SUCCESS {
+		return nil, errors.New("Unable to open query through DLL call")
+	}
+
+	// test path
+	ret = win.PdhValidatePath(counter)
+
+	if ret != win.ERROR_SUCCESS {
+		return nil, errors.New(fmt.Sprintf("Unable to valid path to counter. Error code is %x\n", ret))
+	}
+
+	ret = win.PdhAddEnglishCounter(queryHandle, counter, 0, &counterHandle)
+	if ret != win.ERROR_SUCCESS {
+		return nil, errors.New(fmt.Sprintf("Unable to add process counter. Error code is %x\n", ret))
+	}
+
+	ret = win.PdhCollectQueryData(queryHandle)
+	if ret != win.ERROR_SUCCESS {
+		return nil, errors.New(fmt.Sprintf("Got an error: 0x%x\n", ret))
+	}
+
+	out := make(chan Metric)
+
+	go func() {
```

Comment:
There are lots of utilities for this kind of thing here:

https://github.com/kubernetes/kubernetes/blob/ebf24c14a916adebc37023be69f112d2c97759a9/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L78

`wait.Forever(fn, period)`

is what you want.